oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/zipkin/ZipkinQueryEsDAO.java

```diff
@@ -236,7 +238,9 @@ public List<List<Span>> getTraces(final Set<String> traceIds) {
         BoolQueryBuilder query = Query.bool().must(Query.terms(ZipkinSpanRecord.TRACE_ID, new ArrayList<>(traceIds)));
         SearchBuilder search = Search.builder().query(query).sort(ZipkinSpanRecord.TIMESTAMP_MILLIS, Sort.Order.DESC)
                                      .size(SCROLLING_BATCH_SIZE); //max span size for 1 scroll
-        final SearchParams params = new SearchParams().scroll(SCROLL_CONTEXT_RETENTION);
+        final SearchParams params = new SearchParams()
+                .scroll(SCROLL_CONTEXT_RETENTION)
+                .routing(String.join(",", traceIds));
```

Comment:
@kezhenxu94 Could you check this?

Reply:
One last thing, should we expose API as Iterable rather than a string?

Reply:
as Iterable, the param logic more cohesive.
I change it